src/eth_rpc/servers/kakarot_rpc.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    eth_provider::{provider::EthereumProvider, starknet::kakarot_core::to_starknet_transaction},
+    eth_rpc::api::kakarot_api::KakarotApiServer,
+};
+use jsonrpsee::core::{async_trait, RpcResult};
+use reth_primitives::B256;
+use starknet::{
+    core::{
+        crypto::compute_hash_on_elements,
+        types::{BroadcastedInvokeTransaction, BroadcastedInvokeTransactionV1, FieldElement},
+    },
+    providers::Provider,
+};
+use std::convert::TryInto;
+
+#[derive(Debug)]
+pub struct KakarotRpc<EP, SP> {
+    eth_provider: EP,
+    starknet_provider: SP,
+}
+
+impl<EP, SP> KakarotRpc<EP, SP> {
+    pub const fn new(eth_provider: EP, starknet_provider: SP) -> Self {
+        Self { eth_provider, starknet_provider }
+    }
+}
+trait ToElements {
+    fn try_into_v1(self) -> Result<BroadcastedInvokeTransactionV1, &'static str>;
+}
+
+impl ToElements for BroadcastedInvokeTransaction {
+    fn try_into_v1(self) -> Result<BroadcastedInvokeTransactionV1, &'static str> {
+        match self {
+            Self::V1(tx_v1) => Ok(tx_v1),
+            Self::V3(_) => Err("Transaction is V3, cannot convert to V1"),
+        }
+    }
+}
+
+#[async_trait]
+impl<EP, SP> KakarotApiServer for KakarotRpc<EP, SP>
+where
+    EP: EthereumProvider + Send + Sync + 'static,
+    SP: Provider + Send + Sync + 'static,
+{
+    async fn kakarot_get_starknet_transaction_hash(&self, hash: B256, retries: u8) -> RpcResult<B256> {
+        // Retrieve the stored transaction from the database.
+        let transaction: reth_rpc_types::Transaction =
+            self.eth_provider.transaction_by_hash(hash).await.unwrap().unwrap();
```

Comment:
We should also handle the potential errors here instead of unwrapping no?

Reply:
Fixed, thanks